script/upgrades/eXOF/eXOFChecks.verify.sol

```diff
@@ -0,0 +1,517 @@
+    }
+    console.log("🟢 StableTokenXOFProxy has the correct implementation address");
+  }
+
+  function verifyEXOFAddedToRegistry() internal view {
+    address registryEXOFAddress = IRegistry(REGISTRY_ADDRESS).getAddressForStringOrDie("StableTokenXOF");
+    address deployedEXOFAddress = contracts.deployed("StableTokenXOFProxy");
+
+    if (registryEXOFAddress != deployedEXOFAddress) {
+      console.log(
+        "The eXOF address from the registry: %s does not match the deployed address: %s.",
+        registryEXOFAddress,
+        deployedEXOFAddress
+      );
+      revert("Deployed eXOF does not match what registry points to. See logs.");
+    }
+
+    console.log("🟢 eXOF has been added to the registry");
+  }
+
+  function verifyEXOFAddedToReserves() internal view {
+    if (!Reserve(address(uint160(partialReserve))).isStableAsset(eXOF)) {
+      revert("eXOF has not been added to the partial reserve.");
+    }
+
+    if (!Reserve(address(uint160(reserve))).isStableAsset(eXOF)) {
+      revert("eXOF has not been added to the reserve.");
+    }
+
+    console.log("🟢 eXOF has been added to the reserves");
+  }
+
+  function verifyEXOFAddedToFeeCurrencyWhitelist() internal view {
+    address[] memory feeCurrencyWhitelist = IFeeCurrencyWhitelist(contracts.celoRegistry("FeeCurrencyWhitelist"))
+      .getWhitelist();
+
+    if (!Arrays.contains(feeCurrencyWhitelist, eXOF)) {
+      revert("eXOF has not been added to the fee currency whitelist.");
+    }
+
+    console.log("🟢 eXOF has been added to the fee currency whitelist");
+  }
+
+  function verifyExchanges() internal {
+    eXOFConfig.eXOF memory config = eXOFConfig.get(contracts);
+
+    console.log("\n== Verifying exchanges ==");
+
+    verifyPoolExchange(config);
+    verifyPoolConfig(config);
+    verifyTradingLimits(config);
+  }
+
+  function verifyPoolExchange(eXOFConfig.eXOF memory config) internal view {
+    bytes32[] memory exchanges = BiPoolManager(biPoolManagerProxy).getExchangeIds();
+
+    // check configured pools against the config
+    require(
+      exchanges.length == config.pools.length + 7,
+      "Number of expected pools does not match the number of deployed pools."
+    );
+
+    for (uint256 i = 0; i < config.pools.length; i++) {
+      bytes32 exchangeId = getExchangeId(config.pools[i].asset0, config.pools[i].asset1, config.pools[i].isConstantSum);
+
+      IBiPoolManager.PoolExchange memory pool = BiPoolManager(biPoolManagerProxy).getPoolExchange(exchangeId);
+      Config.Pool memory poolConfig = config.pools[i];
+
+      // verify asset0 of the deployed pool against the config
+      if (pool.asset0 != poolConfig.asset0) {
+        console.log(
+          "The asset0 of deployed pool: %s does not match the expected asset0: %s.",
+          pool.asset0,
+          poolConfig.asset0
+        );
+        revert("asset0 of pool does not match the expected asset0. See logs.");
```

Comment:
maybe let's put statements like these into `require()` instead? like how it's done in a few lines below in L191, L196. This is inconsistent across other functions as well

Reply:
🙅 Consistency isn't the point here. The log statements are here to add additional information using string interpolation to help debug any failed checks. Solidity doesn't support string interpolation in require statements, so it's best this remains this way.